src/corecel/math/Atomics.hh

```diff
@@ -24,20 +25,17 @@ CELER_FORCEINLINE_FUNCTION T atomic_add(T* address, T value)
 {
 #if CELER_DEVICE_COMPILE
     return atomicAdd(address, value);
-#elif defined(_OPENMP)
+#else
     CELER_EXPECT(address);
     T initial;
-#    pragma omp atomic capture
+#    if defined(_OPENMP)
+#        pragma omp atomic capture
+#    endif
```

Comment:
Is there an "expectation" on the type supported by this function?  In both the original and new code, on the CPU (without OpenMP) the routine upholds its name only if the type is `std::atomic<S>`.

Reply:
I.e do we need:
```
#if CELER_DEVICE_COMPILE OR defined(_OPENMP)
template<T>
#else
template<std::atomic<T>>
#endif
```

Reply:
Yeah, I thought the same thing, so I added a `brief` comment to the file to try to address this: these should only be atomic calls for *kernel* functions (i.e. inside an OpenMP loop or CUDA kernel). Regular CPU threading with `std::atomic` shouldn't apply in that case.